readmes/custom_viz.md

```diff
@@ -0,0 +1,289 @@
+# Making a Custom Visualizer
+
+While the Visualization Package provides a preconfigured default visualization suite, there are many compelling cases for custom visualizations. This page steps through how to create a custom visualizer for the Nav2 project that tracks a history of `/goal_pose` messages over time, drawing a line between each point. 
+
+**Table of Contents**
+- [Creating a New Visualizer](#creating-a-new-visualizer)
+    - [Drawing the UI Window](#drawing-the-ui-window)
+    - [Creating Drawings](#creating-drawings)
+- [Now What?](#now-what)
+    - [Exercise the Example](#exercise-the-example)
+    - [Automate It](#automate-it)
+    - [Learn more about this Unity Scene](#learn-more-about-this-unity-scene)
+    - [Import your own robot](#import-your-own-robot)
+    - [Experiment with your own Scenes](#experiment-with-your-own-scenes)
+    - [Learn more about Visualizations](#learn-more-about-visualizations)
+
+> This section assumes you have already set up your environment according to the [configuration guide](dev_env_setup.md) and have run the example successfully as described [here](run_example.md).
+
+---
+
+## Creating a New Visualizer
+
+- If you have the `DefaultVisualizationSuite` in your scene from the previous tutorial, the necessary components are already added for visualization.
+
+    - If you do not have the `DefaultVisualizationSuite`, you can instead use the `EmptyVisualizationSuite`, which contains the necessary extensions for visualizations. To add the empty visualization suite, in the Project window, expand and select `Packages/Robotics Visualization`. Select the `EmptyVisualizationSuite` and drag it into your scene Hierarchy.
+
+- To start writing a custom visualizer, start by making a new script in the `Assets/Scripts` directory named `PoseTrailVisualizer`.
+
+    > To create a new script, right click on `Assets/Scripts` in Unity's Project window, and select `Create > C# Script`.
+
+    > To skip to the end using the completed script, you can add the file from [unity_scripts/PoseTrailVisualizer](../unity_scripts/PoseTrailVisualizer.cs) to `Assets/Scripts`, attach it to a GameObject in your scene, configure the component in the inspector, and run the example.
+
+- In your visualization suite, make a new GameObject named `TrailVisualizer` (e.g. `DefaultVisualizationSuite/TrailVisualizer`). 
+
+    Add the newly created PoseTrailVisualizer component to the TrailVisualizer GameObject. 
```

Comment:
I got an error when I tried to add the script as a component to the GameObject at this state. I continued on to filling out the script and then tried again, and it worked.

Reply:
oh, that is unexpected--what did it tell you the error was? i can add a note that you should let unity finish compiling before you do this--was that similar to the error it gave you?

Reply:
Hmm, I just tried to reproduce it and I couldn't.